src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsaAlgorithm.cs

```diff
@@ -0,0 +1,326 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Security.Cryptography;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    /// <summary>
+    ///   Represents a composite ML-DSA algorithm identifier, combining ML-DSA with a traditional algorithm.
+    /// </summary>
+    [Experimental(Experimentals.PostQuantumCryptographyDiagId, UrlFormat = Experimentals.SharedUrlFormat)]
+    public sealed class CompositeMLDsaAlgorithm : IEquatable<CompositeMLDsaAlgorithm>
+    {
+        /// <summary>
+        ///   Gets the name of the algorithm.
+        /// </summary>
+        /// <value>
+        ///   An a string representing the algorithm name.
+        /// </value>
+        public string Name { get; }
+
+        /// <summary>
+        ///   Gets the maximum signature size in bytes for the composite algorithm.
+        /// </summary>
+        public int MaxSignatureSizeInBytes { get; }
+
+        internal MLDsaAlgorithm MLDsaAlgorithm { get; }
+
+        internal string Oid { get; }
+
+        private CompositeMLDsaAlgorithm(
+            string name,
+            MLDsaAlgorithm mlDsaAlgorithm,
+            int maxTraditionalSignatureSize,
+            string oid)
+        {
+            Name = name;
+            MLDsaAlgorithm = mlDsaAlgorithm;
+            MaxSignatureSizeInBytes = MLDsaAlgorithm.SignatureSizeInBytes + maxTraditionalSignatureSize;
+            Oid = oid;
+        }
+
+        /// <summary>
+        ///   Gets a Composite ML-DSA algorithm identifier for the ML-DSA-44 and 2048-bit RSASSA-PSS with SHA256 algorithm.
+        /// </summary>
+        /// <value>
+        ///   An ML-DSA algorithm identifier for the ML-DSA-44 and 2048-bit RSASSA-PSS with SHA256 algorithm.
+        /// </value>
+        public static CompositeMLDsaAlgorithm MLDsa44WithRSA2048Pss =           new("MLDSA44-RSA2048-PSS-SHA256",
```

Comment:
Should `HashAlgorithmName`/`HashAlgorithmName?` be an arg somewhere?

Reply:
SHAKE isn't a well-known `HashAlgorithmName` and one of the algos uses it. It doesn't prevent us from adding it, but we can do it when we end up needing it.